reactioncommerce/reaction-next-starterkit
# feat: add title and meta data to ProductGrid

Needs https://github.com/reactioncommerce/reaction/pull/4209 to be merged in to work

Issue: #16 
Impact: **minor**
Type: **feat**

## Issue

Add title and description meta tags to the Product Grid page, which change based on if you are on the default landing page, or a tag landing page

## Solution 
- Import `react-helmet` into the Product Grid page file
- add data to mobx
- add dynamic `title` and `meta description` tags via `react-helmet`

## Test

- Start the app, go to `http://localhost:4000`
- See that title and description meta tags are based upon shop data
- Navigate to a tag
- See that the title and description meta tags update based on the tag clicked

## Testing FB & Twitter meta
- import the `TwitterSocial` and `FacebookSocial` components:
```
import { FacebookSocial, TwitterSocial } from "components/Social";
```

Pass meta data into the `settings` prop of either of those components (for example, the `render()` of the product grid):
```
render() {
    const { shop } = this.props;
    const meta = {
      "description": shop.description,
      "siteName": shop.name,
      "title": shop.name
    }

    return (
      <Layout title="Reaction Shop">
        {this.renderHelmet()}
        <FacebookSocial settings={meta} />
        <TwitterSocial settings={meta} />
        <ProductGrid catalogItems={this.props.catalogItems} />
      </Layout>
    );
  }
```

See that OG and twitter meta is now available.

This follows the patter we used in classic reaction, where the Facebook or Twitter metatags are not added until a social button is on a page.
